src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Cancel.cs

```diff
@@ -63,9 +63,16 @@ public async Task<IActionResult> CancelAsync(string workitemInstanceUid)
         if (response.Status is Core.Messages.Workitem.WorkitemResponseStatus.Conflict
             && !string.IsNullOrEmpty(response.Message))
         {
-            Response.Headers.Warning = string.Format(DicomApiResource.WarningHeader, response.Message);
+            Response.SetWarning(Core.Models.HttpWarningCode.MiscPersistentWarning, GetHost(), response.Message);
         }
 
         return StatusCode((int)response.Status.CancelResponseToHttpStatusCode(), response.Message);
     }
+
+    private string GetHost()
+    {
+        string host = Request.Host.Host;
+        // As Dicom standard, should append colon after service. https://dicom.nema.org/medical/dicom/current/output/chtml/part18/sect_11.7.3.2.html
+        return string.IsNullOrWhiteSpace(host) ? host : host + ":";
```

Comment:
Is the colon still present if the host is missing and replaced with `'-'`?

Reply:
yeah, that's the trickly part.  As http standard, it should be `'_'`.